src/prefect/infrastructure/kubernetes.py

```diff
@@ -263,28 +264,45 @@ async def run(
         if not self.command:
             raise ValueError("Kubernetes job cannot be run with empty command.")
 
-        # if a k8s cluster block is provided to the flow runner, use that
-        if self.cluster_config:
-            self.cluster_config.configure_client()
-        else:
-            # If no block specified, try to load Kubernetes configuration within a cluster. If that doesn't
-            # work, try to load the configuration from the local environment, allowing
-            # any further ConfigExceptions to bubble up.
-            try:
-                kubernetes.config.load_incluster_config()
-            except kubernetes.config.ConfigException:
-                kubernetes.config.load_kube_config()
-
+        self._set_config()
```

Comment:
Whoa scary config side-effect function. Let's name this like... "`_configure_kubernetes_client_config` or something — we should also note that this is isn't threadsafe and that we need to do something else eventually.